Add chunk batch window

BuddhiWathsala opened the pull request:
## Purpose
> Resolved issue #853

## Approach
- Take the results of the batch window previously had.
- Then chunk the batch according to user input using reset events.
- Finally, process the chunks. 

## Automation tests
 - Unit tests 
   > Used previous batch tests for the new chunkBatch window with minor changes.

suhothayan commented:
IMHO we should not create another window to achieve this. Can you improve the window.batch() to support the `length` parameter and support this functionality?

mohanvive commented:
Yes, it should be an improvement to the existing window as mentioned in the issue. @BuddhiWathsala Can you please improve the existing batch window with your above implementation?

BuddhiWathsala commented:
Removed the chunkBatch window that I previously implemented. Added the relevant changes to the batch window accordingly. @tishan89 can you please review this?

tishan89 commented on modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/BatchWindowProcessor.java:
```diff
@@ -120,12 +141,51 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
                     } while (streamEventChunk.hasNext());
                 }
-                resetEvent = streamEventCloner.copyStreamEvent(streamEventChunk.getFirst());
+                streamEventChunkStore.add(streamEventChunk.getFirst());
```
Why do you need to add this to a new chink. Can't we use the existing streamEventChunk

tishan89 commented on modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/BatchWindowProcessor.java:
```diff
@@ -120,12 +141,51 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
+                        if (resetEvent != null) {
+                            currentEventChunk.add(resetEvent);
+                        }
+                        resetEvent = null;
```
We do not need this null assignment

tishan89 replied:
May be this should go into else clause of above if block

tishan89 commented on modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/BatchWindowProcessor.java:
```diff
@@ -120,12 +141,51 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
+                if (resetEvent != null) {
+                    currentEventChunk.add(resetEvent);
+                }
+                resetEvent = null;
```
unwanted null assignment

tishan89 commented on modules/siddhi-core/src/test/java/org/wso2/siddhi/core/window/BatchWindowTestCase.java:
```diff
@@ -240,6 +240,63 @@ public void receive(long timestamp, Event[] inEvents, Event[] removeEvents) {
+            public void receive(long timestamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timestamp, inEvents, removeEvents);
+                if (inEvents != null) {
+                    inEventCount = inEventCount + inEvents.length;
```
Please assert the count here also

BuddhiWathsala commented:
Added reviewed requests by @tishan89 .

tishan89 commented on README.md:
```diff
@@ -30,19 +30,6 @@ Siddhi supports the following:
 
 Siddhi is free and open source, under **Apache Software License v2.0**.
 
-## Table of contents
```
Why are you removing this

tishan89 commented:
You haven't merged properly. Changes in master will be overwritten by this PR. Can you please check.
